### DISCUSSION TITLE
#12 📑 Weekly Report 12

### DISCUSSION THREAD

**Discussion by ins305** [SE-TINF22B6/Statify#89] [Blog]
Hello everyone👋 

## Testing
Last week was all about tests. As our testing frameworks, we are using Jest for our React Frontend and JUnit for the Java Spring Boot Backend. We had already set up these frameworks, so we can get started with writing more tests now. Currently there's a few unit tests already written:
[Frontend Tests](https://github.com/SE-TINF22B6/Statify/blob/develop/web_app/src/App.test.js)
[Backend Tests](https://github.com/SE-TINF22B6/Statify/tree/develop/backend_service/src/test/java/fm/statify/backend_service)
To ensure that all our code works correctly, we still need to add more tests to get a higher test coverage.

Also, we integrated our tests with Github Actions, so that every commit triggers all tests to be run. Afterwards, you can also see the statistics about how many tests succeeded and failed etc.
The part in our workflows responsible for the reporting of the test results uses the [Test Reporter Github Action](https://github.com/marketplace/actions/test-reporter) and looks like this:
![image](https://github.com/SE-TINF22B6/Statify/assets/64202248/96e7539f-b4c7-4fa6-8c4b-3ebafa28a96b)
The produced test results look like this, when all the tests have passed:
![image](https://github.com/SE-TINF22B6/Statify/assets/64202248/fcfa1797-85ea-4df0-8ff7-0b8f68ac583a)

Another detail that was added are these two small badges in our README that indicate whether the workflows have run successfully.
[![Node.js CI](https://github.com/SE-TINF22B6/Statify/actions/workflows/node.js.yml/badge.svg?branch=develop)](https://github.com/SE-TINF22B6/Statify/actions/workflows/node.js.yml)
[![Java CI with Maven](https://github.com/SE-TINF22B6/Statify/actions/workflows/maven.yml/badge.svg?branch=develop)](https://github.com/SE-TINF22B6/Statify/actions/workflows/maven.yml)

**Comment by NicolasVien** [comment]
Great job on setting up your testing frameworks, Jest for the React Frontend and JUnit for the Java Spring Boot Backend! It's good to hear that you've already written a few unit tests! Keep up the good work on writing and automating your tests to ensure the quality and reliability of your code!

**Comment by BenjaminB109** [comment]
Dear Statify Team, 
I'm glad to see that you have managed to implement the test. Keep up the good work.

**Comment by sfrancisw** [comment]
Awesome work on the testing part! It's great to see the  frontend and backend testing, respectively. The addition of badges in the README adds a nice touch, giving everyone a quick glance at the status of the workflows. Keep up the fantastic work! 
Your team time2shine

**Comment by HowDoIUseGit** [comment]
Great job on implementing testing for both frontend and backend! It's fantastic to see the setup using Jest for the React frontend and JUnit for the Java Spring Boot backend. The addition of badges in the README is a nice touch for quickly checking the status of workflows. Keep up the fantastic work in writing and automating tests to ensure code quality and reliability!
Your CookPit
